packages/openapi-parser/src/utils/join/join.test.ts

```diff
@@ -0,0 +1,982 @@
+import { join } from '@/utils/join/join'
+import { describe, expect, it } from 'vitest'
+
+describe('join', () => {
+  it('should handle joining info objects, prioritizing the first input document', async () => {
+    const result = await join([
+      {
+        info: {
+          title: 'Main title',
+          version: '1.1.1',
+        },
+      },
+      {
+        info: {
+          title: 'Second title',
+          version: '1.43.1',
+          summary: 'Some extra information about the API',
+        },
+      },
+      {
+        info: {
+          title: 'Third title',
+          version: '1.3.1',
+          summary: 'This will be overridden',
+        },
+      },
+    ])
+
+    expect(result).toEqual({
+      ok: true,
+      document: {
+        info: {
+          title: 'Main title',
+          version: '1.1.1',
+          summary: 'Some extra information about the API',
+        },
+        paths: {},
+        webhooks: {},
+        tags: [],
+        servers: [],
+        components: {},
+      },
```

Comment:
can we make it so, it doesn’t add webhooks, tags, servers and components even if they are empty?

according to the specification the required attributes are

* `openapi`
* `info.title`
* `info.version`
* `paths`